### DISCUSSION TITLE
Split conversation into multiple files

### DISCUSSION THREAD

**Discussion by ItalyPaleAle** [ReagentX/imessage-exporter#388] [Q&A] [answered]
Thanks so much for this app which works amazingly well!

I've just tried exporting my entire iMessage history and it is done well. The only issue I've found is that I have some conversations that contain many thousands of messages, and when exported as HTML the file is too large for browsers to render it properly. For example, the conversation with my wife is a 76MB HTML page, which Firefox and Safari have trouble opening, and renders really slowly on Edge.

Would it make sense for imessage-exporter to have a way to automatically split large HTML conversations into multiple files?

**Comment by ReagentX** [comment, marked as answer]
You can use `--start-date` and `--end-date` to resize message exports for any arbitrary date range you want. For more info, see the instructions [here](https://github.com/ReagentX/imessage-exporter/blob/develop/imessage-exporter/README.md#how-to-use).

**Reply by kylejohnsonkj** [reply]
I came to ask the same thing and found this post. The start/end date flags are great, but from my testing HTML files over 10mb are basically broken when you try to scroll. I think it would be a good addition to split up conversations into 10mb batches by default.
